### PR TITLE
Reduced motion in docs - add warning in examples and make horse static

### LINKED ISSUE #5549: Examples are broken in docs
### Description

In the latest versions of Chrome and Safari on MacOS, none of the examples animate.

Eg: https://docs.swmansion.com/react-native-reanimated/docs/animations/withSpring

### Steps to reproduce

Go to https://docs.swmansion.com/react-native-reanimated/docs/animations/withSpring, press the replay button.

### Snack or a link to a repository

https://github.com/software-mansion/react-native-reanimated

### Reanimated version

3.6.1

### React Native version

0.73.1

### Platforms

Web

### JavaScript runtime

None

### Workflow

None

### Architecture

None

### Build type

Release app & dev bundle

### Device

Real device

### Device model

_No response_

### Acknowledgements

Yes

### PR DESCRIPTION
**Pull request by Latropos**
## Summary
Closes #5549
Changes:
* If user has reduced motion preferences on, then they should see the warning
* If user has reduced motion preferences on, the horse should not gallop.
  * The frame for the horse was chosen to match the logo

|Color mode|Screenshot|
|--|--|
|☀️|image|
|🌙|image|

|Color mode|Screenshot|
|--|--|
|☀️|image|
|🌙|image|

## Test plan
I've checked, that I can see these warnings, when my reduced motion is on

### CODE REVIEW

File: docs/src/components/ReducedMotionWarning/styles.module.css [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+.badge {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  font-size: 14px;
+  font-weight: 500;
+  padding: 0.125rem 0.375rem;
+  background-color: var(--swm-admonition-color-caution);
+}
```
Comment (kacperkapusciak) [inline docs/src/components/ReducedMotionWarning/styles.module.css:14]: It looks like you didn't run prettier 🙏

**Review by kacperkapusciak — CHANGES_REQUESTED** [review]

File: docs/src/components/InteractiveExample/index.tsx [hunk 2]
```diff
@@ -1,11 +1,12 @@
 import React from 'react';
-
 import clsx from 'clsx';
 import { useCopyToClipboard } from 'usehooks-ts';
 
 import BrowserOnly from '@docusaurus/BrowserOnly';
 import CodeBlock from '@theme/CodeBlock';
 import AnimableIcon, { Animation } from '@site/src/components/AnimableIcon';
+import { useReduceMotion } from 'react-reduce-motion';
```
Comment (piaskowyk) [inline docs/src/components/InteractiveExample/index.tsx:8]: We have own hook `useReduceMotion` 🫣
- https://github.com/software-mansion/react-native-reanimated/blob/main/src/reanimated2/hook/useReducedMotion.ts
- https://github.com/software-mansion/react-native-reanimated/blob/main/src/reanimated2/PlatformChecker.ts#L52-L59

Comment (kacperkapusciak) [reply docs/src/components/InteractiveExample/index.tsx:8]: Let's use that

File: docs/src/components/ReducedMotionWarning/index.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+export default function ReducedMotionWarning() {
+  const { colorMode } = useColorMode();
+  return (
+    <div className={styles.badge}>
+      <div className={styles.dangerMark}>
+        {colorMode === 'light' ? <Danger /> : <DangerDark />}
+      </div>
+      Look like reduced motion is turned on in your system preferences, some of
+      the animations may be skipped.
```
Comment (kacperkapusciak) [inline docs/src/components/ReducedMotionWarning/index.tsx:16]:
```suggestion
      It looks like reduced motion is turned on in your system preferences. Some of
      the animations may be skipped. [Learn more](https://docs.swmansion.com/react-native-reanimated/docs/guides/accessibility#reduced-motion-in-animations).
```

File: docs/src/components/InteractiveExample/index.tsx (hunk 2 above)
Comment (Latropos) [reply docs/src/components/InteractiveExample/index.tsx:8]: Our `useReducedMotion` requires refreshing the page to detect accessibility changes. Is it ok?

Comment (Latropos) [reply docs/src/components/InteractiveExample/index.tsx:8]: On the other hand the warning would be compatible with actual animation, so its fine

File: docs/src/components/ReducedMotionWarning/index.tsx (hunk 3 above)
Comment (Latropos) [reply docs/src/components/ReducedMotionWarning/index.tsx:16]: There should be no "the" before "reduced motion", should it?

Comment (kacperkapusciak) [reply docs/src/components/ReducedMotionWarning/index.tsx]: I'm not a native speaker but Grammarly says it's fine without the "the"

**Comment by kacperkapusciak** [comment]
the PR isn't ready for my rereview, please all the comments first

File: docs/src/components/ReducedMotionWarning/index.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    <div className={styles.container}>
+      <div className={styles.dangerMark}>
+        {colorMode === 'light' ? <Danger /> : <DangerDark />}
+      </div>
+      <p className={styles.warningText}>
+        It looks like reduced motion is turned on in your system preferences.
+        Some of the animations may be skipped.{' '}
+        <Link
+          className={styles.link}
+          to="https://docs.swmansion.com/react-native-reanimated/docs/guides/accessibility#reduced-motion-in-animations">
```
Comment (kacperkapusciak) [inline docs/src/components/ReducedMotionWarning/index.tsx:21]:
```suggestion
          to="/docs/guides/accessibility#reduced-motion-in-animations">
```
we prefer relative links wherever possible

Comment (Latropos) [reply docs/src/components/ReducedMotionWarning/index.tsx]: Fixed, opens in a new window

File: docs/src/components/InteractiveExample/index.tsx [hunk 5]
```diff
@@ -37,13 +38,18 @@ export default function InteractiveExample({
...
   return (
     <BrowserOnly fallback={<div>Loading...</div>}>
       {() => (
         <div
           className={`${styles.container} ${larger && styles.largerContainer} 
           ${!showPreview ? styles.code : ''}`}
           data-ispreview={showPreview}>
+          {!!showPreview && !!prefersReducedMotion ? (
+            <ReducedMotionWarning />
+          ) : null}
```
Comment (piaskowyk) [inline docs/src/components/InteractiveExample/index.tsx:52]:
```suggestion
          {(showPreview && prefersReducedMotion) && <ReducedMotionWarning />}
```

File: docs/src/components/InteractivePlayground/index.tsx [hunk 6]
```diff
@@ -54,10 +56,13 @@ export default function InteractivePlayground(
     resetOptions();
   };
 
+  const prefersReducedMotion = useReducedMotion();
+
   return (
     <BrowserOnly fallback={<div>Loading...</div>}>
       {() => (
         <div className={styles.container}>
+          {!!prefersReducedMotion ? <ReducedMotionWarning /> : null}
```
Comment (kacperkapusciak) [inline docs/src/components/InteractivePlayground/index.tsx:65]:
```suggestion
          {prefersReducedMotion && <ReducedMotionWarning /> }
```

**Review by kacperkapusciak — APPROVED** [review]
🥳 🎉 🥳

**Pull request merged**